DOCUMENTATION.md

```diff
@@ -0,0 +1,335 @@
+
+## Table of Contents
+
+- [General Information](#general-information)
+- [Prerequisites](#prerequisites)
+    - [1. Install Docker](#1-install-docker)
+    - [2. Create a GitHub Access Token](#2-create-a-github-access-token)
+    - [3. Install Node via nvm](#3-install-node-via-nvm)
+- [Setup](#setup)
+- [Custom NPM Commands](#custom-npm-commands)
+    - [Start All Services](#start-all-services)
+    - [Start an Individual Service](#start-an-individual-service)
+    - [Logs](#logs)
+    - [Stop All Services](#stop-all-services)
+- [Coding Conventions](#coding-conventions)
+    - [HTML](#html)
+        - [Class Order](#class-order)
+    - [CSS](#css)
+    - [Testing](#testing)
+    - [Commit Messages](#commit-messages)
+    - [Branches](#branches)
+- [Useful pm2 Commands](#useful-pm2-commands)
+    - [Show All Running Services](#show-all-running-services)
+    - [Synchronize Running Services](#synchronize-running-services)
+    - [Show Logs](#show-logs)
+    - [Stop All Services](#stop-all-services)
+    - [Stop Individual Service](#stop-individual-service)
+    - [Delete All Services](#delete-all-services)
+    - [Delete Individual Service](#delete-individual-service)
+
+<hr style="color:white; height:1.5px; width:100%;"/>
+
+
+## General Information
+
+Technically, a11yphant is split into three applications:
+
+1. The **API** (Backend)
+2. The **Site** (Frontend)
+3. The **Submission Checker** (Backend)
+
+All three applications are written in [Typescript](https://www.typescriptlang.org). 
+The **Site** uses the React framework [NextJS](https://nextjs.org), [GraphQL](https://graphql.org/) and [Tailwind](https://tailwindcss.com/).
+The **API** and the **Submission Checker** use the Node framework [NestJS](https://nestjs.com). The **API** also uses [Prisma](https://www.prisma.io) and GraphQL. [PostgreSQL](https://www.postgresql.org/) is used as the database.
+
+## Prerequisites
+
+### 1. Install Docker
+
+This project uses [Docker](https://www.docker.com/). Please make sure you have it installed.
+
+### 2. Create a GitHub Access Token
+
+Create an access token for the Github API. This access token is required to install our NPM packages from the GitHub NPM registry. You can follow [this guide](https://docs.github.com/en/authentication/keeping-your-account-and-data-secure/creating-a-personal-access-token).
+
+**TLDR:**
+
+1. Log into your GitHub account.
+2. Go to *Settings* -> *Developer settings* -> *Personal access tokens* -> *Tokens (classic)*.
+3. Create a token and choose the scope `read:packages`.
+4. In your terminal, run the following command and login with your github user and the just created token as password:
+```bash
+npm login --scope=@a11yphant --auth-type=legacy --registry=https://npm.pkg.github.com
+```
+
+### 3. Install Node via `nvm`
+
+This is important. You must have Node installed via `nvm` for this project to work correctly ([Setup Instructions](https://github.com/nvm-sh/nvm#installing-and-updating)).
```

Comment:
It's actually more of an optional step. As long as the node version that is specified in the .nvmrc file is used it does not matter what is used to setup up the correct version. Even uninstalling node and manually installing the correct version would work. Using nvm is more like a convenience thing.